core-primitives/enclave-api/src/remote_attestation.rs

```diff
@@ -373,6 +385,48 @@ impl RemoteAttestation for Enclave {
 				collateral_ptr_ptr,
 			)
 		};
+
+		unsafe {
+			trace!("FMSPC: {:?}", hex::encode(fmspc));
+			let collateral = &*collateral_ptr;
```

Comment:
This is the only _truly_ `unsafe` operation, how about moving everything else out of the `unsafe` block?
Like so:
``` rust
let collateral = unsafe { &*collateral_ptr };
trace!("collateral:");
trace!(...);
```

Reply:
Yes, much better!

Reply:
As discussed during the daily, this needs to merged ASAP, that is why I pushed those commits without asking for permission before, sorry about this @coax1d.

I checked it locally and by default it printed the memory addresses only, not the values stored at them. (Why there is a Rust struct which has C style fields, but no convince method for using it is a mystery to me :cry:)
This behaviour was changed in 99cdba9, as I think this was the intention.